Zend/zend_vm_def.h

```diff
@@ -5181,8 +5181,14 @@ ZEND_VM_C_LABEL(num_index):
 			str = ZSTR_EMPTY_ALLOC();
 			ZEND_VM_C_GOTO(str_index);
 		} else if (Z_TYPE_P(offset) == IS_DOUBLE) {
-			hval = zend_dval_to_lval(Z_DVAL_P(offset));
-			ZEND_VM_C_GOTO(num_index);
+			if (Z_DVAL_P(offset) <= INT_MAX) {
+				hval = zend_dval_to_lval(Z_DVAL_P(offset));
+				ZEND_VM_C_GOTO(num_index);
+			} else {
+				convert_to_string(offset);
```

Comment:
This changes the value of the key. If this is a variable, then it will change it. E.g.
```php
$key = 9999999999999999999.0;
$a = [$key => 1];
var_dump($a, $key);
```

This will now output:
```
array(1) {
  ["1.0E+19"]=>
  int(1)
}
string(7) "1.0E+19"
```